### ISSUE TITLE
Can we talk about buttons...

### ISSUE THREAD

**Issue by Dechowmedia** [youseedk/dna#246]
Currently the documentation written here:

https://dna.yousee.dk/docs/components/form-elements/buttons.html

Indicates that buttons should be used as "links" - however there are two issues with this:
- The documentation is clearly within the "Form Elements" part of DNA, AFAIK, a link is not a form element.
- Why would we want that, when we have had a major discussion in regards to the `cursor: pointer` being used at all on links? (when they were buttons)

My suggestions:
- Links as Buttons are either **banned** (we are already not using it in dna-react-jss at least in the "button component")
- If button links are to be used - it should be indicated as such in another section more appropriate, while having the DOM for an ACTUAL button be present in the documentation

**Comment by havgry** [comment]
I agree with you regarding the documentation. The following improvements could easily be made:

1. Rename "form elements" into "input elements", "data entry" or something like that and move the button component out of that category and into a more general category.
2. Expand the documentation to clearly cover both `button` and `a` variants.

However, while I generally agree with the sentiment that we should refrain from using buttons for navigational purposes, this distinction can be difficult to work with for a UX/UI designer. When is it a new page and when is it merely a new state in the same page/component?

A brief glance at the frontpage of Yousee.dk further reveals that we use buttons as links in 3 out of 5 of the main call to actions. It shouldn't be an argument for rejecting a proposal but "banning buttons as links" is a UX first decision, I'd say.

**Comment by JesperHoff** [comment]
> It shouldn't be an argument for rejecting a proposal but "banning buttons as links" is a UX first decision, I'd say.

Agreed. It should still be possible to do `<a>` buttons. I have an example where the button generates a calendar url, but for android phones, that is generated by forwarding to `https://calendar.google.com/calendar/render/....` where as for iphones, its generating an .ics that they then import. Having one as a link and one as a button would just look horrible, so UX > technicalities => the need to handle this on a case by case basis.

**Comment by Dechowmedia** [comment]
I am not for banning buttons as links either - just that we had this exact discussion (at a Frontend guild long ago - in a galaxy far far away...) about buttons as links, when the `cursor: pointer` idiom came up.

So I wanted to retire that discussion as the DNA docs (now) clearly states it's okay ;) 

Should we assign someone to fix-up the docs? I can do so but only after this week (or later on) after my handover is done.

**Comment by havgry** [comment]
I think we should split this up into 2 PRs: One updating the docs for the Button component defaulting to `button` examples perhaps with a single `a` example, and one doing the restructuring of the Button component and potentially all the other form element components.

Any thoughts, @grandorf?

**Comment by grandorf** [comment]
If you use `<a class="ys-button">` the button will have the pointer as cursor. Because it is a link.

If you use `<button class="ys-button">` the button will NOT have the pointer as cursor. Because it is not a link.

So don't get confused about this part.

Regarding the documentation page I agree we can improve this. The <a> element is not a form element and maybe we should move the  buttons to the [Link](https://dna.yousee.dk/docs/components/text/link) section?

For me it doesn't really matter if we list the buttons as <a> or <button> elements. We do say in the guidelines that the classes can be used for both:

**HTML Guidelines**

- Only <a> and <button> elements are allowed to use the .button class
- Remember to specify the type attribute when using a button element. The default is submit but in most cases you probably want to use type="button"

**Comment by grandorf** [comment]
One more small bug: The class name is not `.button`. It's `ys-button` and should be updated on the page under HTML guidelines :)

**Comment by grandorf** [comment]
Have we talked about buttons? Can I close the issue? I will change the text regarding my last comment.

**Comment by Dechowmedia** [comment]
Sure :) - Make a PR with the changes then I will approve and we can be on our marry way :)
